src/Stratis.Bitcoin.Tests/Consensus/ChainedHeaderTreeTest.cs

```diff
@@ -196,6 +208,46 @@ public void ConnectHeaders_SupplyHeadersThenSupplyMore_Both_Tip_PeerId_Maps_Shou
             Assert.True(peerTipsByPeerIdBefore.Values.Count == peerTipsByPeerIdAfter.Values.Count);
         }
 
+        /// <summary>
+        /// Issue 5 @ Create chained header tree component #1321
+        /// 2 peers should supply headers - one of the peers creates a fork.
+        /// Make sure that ChainedBlock is not created more than once for the same header.
+        /// Check that next pointers were created correctly.
+        /// </summary>
+        [Fact]
+        public void ConnectHeaders_HeadersFromTwoPeersWithFork_ShouldCreateBlocksForNewHeaders()
```

Comment:
This test doesn't test what should be tested. 

Original test description is: 

```
5) 3 peers should supply headers so one of the peers is on a fork. Make sure that ChainedBlock is not created more than once for the same header. Check that next pointers were created correctly
```

So you have 3 peers that present you this: 
```
1) 1a-2a-3a-4a-5a-6a-7a-8a
2) 1a-2a-3a-4a-5a-6a-7a-8a
3) 1a-2a-3a-4a-5b-6b-7b-8b
```

After that CHT should contain 12 headers:

```
1a-2a-3a-4a-5a-6a-7a-8a
           -5b-6b-7b-8b
```

and each of them should have 1 .Next pointer except for 8a and 8b which contain none and 4a which contains 2

Reply:
Changed accordingly